src/a4/domain/Property.java

```diff
@@ -1,5 +1,36 @@
 package a4.domain;
 
-public class Property {
+public class Property extends PropertySpace {
+	int value;
```

Comment:
I made it extend PropertySpace to use the Factory. We can change this if you think it is better.